src/AutorestSwift/View Models/ReturnTypeViewModel.swift

```diff
@@ -38,21 +38,25 @@ struct ReturnTypeViewModel {
     init(from responses: [ResponseViewModel]?) {
         var objectTypes = Set<String>()
         var strategies = Set<String>()
+        var isNullables = Set<Bool>()
         for response in responses ?? [] {
             if response.strategy != "noBody" {
                 objectTypes.insert(response.objectType)
             }
             strategies.insert(response.strategy)
+            isNullables.insert(response.isNullable)
         }
 
         // Only supprt at most 1 body or pageBody return type now.
         let withBodyStrategies = strategies.filter { $0 != "noBody" }
-        let hasNoBodyStrategies = strategies.contains { $0 == "noBody" }
         assert(withBodyStrategies.count <= 1)
 
+        let hasNoBodyStrategies = strategies.contains { $0 == "noBody" }
+        let hasNullableResponse = isNullables.contains { $0 == true }
```

Comment:
Hmm, why not something like:

```Swift
var hasNullableResponse = false
for response in responses ?? [] {
  hasNullableResponse = hasNullableResponse || response.isNullable
}
```
Or
```Swift
let hasNullableResponse = responses.first(where: { $0.isNullable == true }) != nil
```

Reply:
If `strategies` isn't used elsewhere except to set the value of `hasNoBodyStrategies` then I'd recommend a similar approach for that too.

Reply:
strategies Set is used to check in the assert to check if there is more than 1 body/pagedBody.  We can remove that once we can remove that assert. So leaving that as is for now.